### PR TITLE
Remove some image configs from azenvtypes that AgentBaker doesn't need

### PR DESCRIPTION
**Pull request by yizhang4321**

### CODE REVIEW

File: cmd/generate.go [hunk 1]
```diff
@@ -80,11 +80,10 @@ func newGenerateCmd() *cobra.Command {
 				},
 
 				OSImageConfig: map[datamodel.Distro]datamodel.AzureOSImageConfig{
-					datamodel.Ubuntu:         datamodel.Ubuntu1604OSImageConfig,
-					datamodel.Ubuntu1804:     datamodel.Ubuntu1804OSImageConfig,
-					datamodel.Ubuntu1804Gen2: datamodel.Ubuntu1804Gen2OSImageConfig,
```
Comment (xuto2) [inline cmd/generate.go:85]: we need this in AKS. But i'm not sure if we need the whole osImageConfig for agentbaker. So if you keep osImageConfig, please keep Ubuntu1804Gen2

File: pkg/agent/datamodel/azenvtypes.go [hunk 2]
```diff
@@ -113,30 +113,6 @@ var (
 		ImageVersion:   "latest",
 	}
 
-	//Ubuntu1804Gen2OSImageConfig is Gen2 flavor the Ubunutu 18.04-LTS Linux distribution.
-	Ubuntu1804Gen2OSImageConfig = AzureOSImageConfig{
-		ImageOffer:     "aks",
-		ImageSku:       "aks-ubuntu-1804-gen2-2020-q3",
-		ImagePublisher: "microsoft-aks",
-		ImageVersion:   "2020.08.06",
-	}
```
Comment (xuto2) [inline pkg/agent/datamodel/azenvtypes.go:122]: we need this.

Comment (yizhang4321) [reply pkg/agent/datamodel/azenvtypes.go:122]: I'll abandon this PR if we still need the code, but before I do it, where are we using it in AgentBaker?

Comment (xuto2) [reply pkg/agent/datamodel/azenvtypes.go:122]: No I don't think we need the OSImageConfig map, but **if we keep it**, we should keep this Ubuntu1804Gen2OSImageConfig along with other four that you keep. It's for completeness.

If you remove the whole OSImageConfig map and all the image configs, I think that will also work

File: cmd/generate.go (hunk 1 above)
Comment (yizhang4321) [reply cmd/generate.go:85]: I removed the whole OSImageConfig

File: pkg/agent/datamodel/azenvtypes.go (hunk 2 above)
Comment (yizhang4321) [reply pkg/agent/datamodel/azenvtypes.go:122]: I removed the whole "OSImageConfig" map since AgentBaker doesn't need it.

**Pull request merged**